-1

Tested at: a036b3ca4250eb4ef770dd95b04c8a4033d78842

CMSSW: CMSSW_10_6_X_2020-11-11-1100
SCRAM_ARCH: slc7_amd64_gcc700
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26eabf/10649/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/BTagSFProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/VertexTableProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/NanoAODOutputModule.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc: In member function 'void GenWeightsTableProducer::fillLHEPdfWeightTablesFromGenInfo({anonymous}::Counter*, const {anonymous}::DynamicWeightChoiceGenInfo*, double, const GenEventInfoProduct&, std::unique_ptr<nanoaod::FlatTable>&, std::unique_ptr<nanoaod::FlatTable>&, std::unique_ptr<nanoaod::FlatTable>&, std::unique_ptr<nanoaod::FlatTable>&) const':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc:483:46: error: 'psWeightDocStr' was not declared in this scope
                             wPS.size() > 1 ? psWeightDocStr : "dummy PS weight (1.0) ",
                                              ^~~~~~~~~~~~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc:483:46: note: suggested alternative: 'psWeightsDocStr'
                             wPS.size() > 1 ? psWeightDocStr : "dummy PS weight (1.0) ",
                                              ^~~~~~~~~~~~~~
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/LeptonJetVarProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenJetTauTaggerProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/PileupJetIDVarProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/EGMEnergyVarProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/VIDNestedWPBitmapProducer.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc:483:46: error: use of undeclared identifier 'psWeightDocStr'; did you mean 'psWeightsDocStr'?
                            wPS.size() > 1 ? psWeightDocStr : "dummy PS weight (1.0) ",
                                             ^~~~~~~~~~~~~~
                                             psWeightsDocStr
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_10_6_X_2020-11-11-1100/src/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc:469:17: note: 'psWeightsDocStr' declared here
    std::string psWeightsDocStr;
</pre>

--- a/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc
+++ b/PhysicsTools/NanoAOD/plugins/GenWeightsTableProducer.cc
@@ -478,11 +478,8 @@ public:
         "", wPDF, weightChoice->pdfWeightsDoc, nanoaod::FlatTable::FloatColumn, lheWeightPrecision_);
 
     outPS = std::make_unique<nanoaod::FlatTable>(wPS.size(), "PSWeight", false);
-    outPS->addColumn<float>("",
-                            wPS,
-                            wPS.size() > 1 ? psWeightDocStr : "dummy PS weight (1.0) ",
-                            nanoaod::FlatTable::FloatColumn,
-                            lheWeightPrecision_);
+    outPS->addColumn<float>("", wPS, psWeightsDocStr, nanoaod::FlatTable::FloatColumn,
+                                    lheWeightPrecision_);
 
     outNamed = std::make_unique<nanoaod::FlatTable>(1, "LHEWeight", true);
     outNamed->addColumnValue<float>(
@@ -1014,7 +1011,7 @@ public:
           if (keepAllPSWeights_) {
             weightChoice->psWeightIDs.push_back(weightIter);  // PS variations
           } else if (std::regex_search(line, groups, mainPSw)) {
-            if (weightChoice->psWeightIDs.size() == 0)
+            if (weightChoice->psWeightIDs.empty())
               weightChoice->psWeightIDs = std::vector<unsigned int>(4, -1);
             int psIdx = (line.find("fsr") != std::string::npos) ? 1 : 0;
             psIdx += (groups.str(2) == "Hi" || groups.str(2) == "_up" || groups.str(2) == "2.0") ? 0 : 2;

